### DISCUSSION TITLE
Why these errors? Any Way to Fix It?

### DISCUSSION THREAD

**Discussion by CableSix** [upscayl/upscayl#388] [Releases]
![image](https://github.com/upscayl/upscayl/assets/120307624/b49c6971-eb5b-4c6f-99ba-b4bc4b654aed)
![image](https://github.com/upscayl/upscayl/assets/120307624/2c69f101-063f-4414-ba49-de219deb61ee)

I converted the images that gave this error with the XnConvert program and it worked in one but in another it gave another decoding error, until the second one gives you the chance to try other Upscaling Types, but in the case of the first one you don't even have that chance, then just converting the image too.

**Comment by CableSix** [comment]
doing tests I discovered some things, the first error must be because the folder of the images that I tried to upscale, they have modified permissions, it was not the conversion that fixed it, but because it created the image outside the source folder. The second error is related to the GPU ID, I removed the ID that I put in and it seems that everything is fine now. Hope this helps someone if any of these errors occur.

**Reply by sagraplus** [reply]
Hi CableSix. I am new to upscayl and I am having the same problem you had. Hopefully you can help me. I saw how you resolved the problem, but though I tried to research it I have not been able to understand folder permissions and GPU IDs and how to change them. I have a Windows 10 Desktop, Intel Quad core, 16 gig ram. I use it heavily (music, graphics, etc) including many ope source programs but i have never had a problem like this. Your help would be appreciated. Bob

**Reply by aaronliu0130** [reply]
> Yes, unfortunately. NCNN Vulkan requires a Vulkan-compatible GPU. Upscayl won't work with most iGPUs or CPUs.

A workaround exists for Windows and Linux in #390.

If you had filled an issue, you'd have seen instructions to read the FAQ.

![compulsory rickrolls are issued for duplicates of issues and the FAQ](https://user-images.githubusercontent.com/11874211/218282922-efab6795-84af-4cfb-9400-c28c8729ea78.gif)

**Reply by sagraplus** [reply]
Hi Aaron. OK, I deleted vulkan-1.dll and renamed vk_swiftshader.dll to vulkan-1.dll. That's what #390 recommends, right? But I don't have a clue what GPU ID is and how to remove it or set it to 0. Please explain. Thanks.

**Reply by aaronliu0130** [reply]
Read the wiki, as filing an issue would've told you. The ID is blank by default.

**Reply by sagraplus** [reply]
I have read everything but still confused. I keep reading contradictory posts and comments. "the #390 is old and has been replaced", you cannot delete vulkan-1.dll or vk_swiftshader.dll, you must  rename a copy of it". Could you please be kind enough to give me a step by step guide?

**Reply by aaronliu0130** [reply]
Hmm, apologies for sending the wrong link. It should be the top post in #580.

**Reply by aaronliu0130** [reply]
Nowhere says that you can't delete vulkan-1.dll though.

**Comment by aaronliu0130** [comment]
Please use issues for bug reports (and include logs). Issues are also preferred for enhancements.

**Reply by sagraplus** [reply]
How do you do that??

**Reply by aaronliu0130** [reply]
The bug report guide is pinned in discussions.

**Comment by sagraplus** [comment]
I just cannot get some step by step help with this. This is lunacy. I read everything on the linked pages I was directed to and still am confused. I am just not very smart. One last plea for step by step help. I'm stunned that this is so complicated

**Reply by aaronliu0130** [reply]
It's in [the wiki](https://github.com/upscayl/upscayl/wiki/Troubleshooting).

**Comment by sagraplus** [comment]
Thank you for trying to help. I have finally worked it out. Aaron, in an email told me the post with the correct instructions is #580, not #390.  I am upscaling an image now. It's only a 99k 700 x 800 image but it looks like it will take about an hour, according to the progress icon. No way around that very long processing time??

**Reply by aaronliu0130** [reply]
Yes it will, because you don't have a GPU.
